### PR TITLE
Updating latest depencies

### PR DESCRIPTION
**Pull request by nikhiljindal**
First commit auto generated by running `glide update --strip-vendor`.

Had to clear cache as per @G-Harmon's suggestion: https://github.com/Masterminds/glide/issues/592#issuecomment-344769706 

Also removed the kubernetes pin that was added in https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/93

Second commit is to fix the build using latest dependencies (renames and code moves).
It has following changes that were required to fix the build:
* Rename ServicePort.Port to Nodeport: kubernetes/ingress-gce#128
* Update the return argument from ListGlobalForwardingRules to be []*ForwardingRule instead of ForwardingRuleList
* Update the import path for constants from ingress-gce like AppProtocol, ProtocolHTTP to be pkg/annotations instead of utils.
* Update the usage of annotations library from ingress-gce to use annotations.FromIngress(ing) instead of annotations.IngAnnotations(ing.ObjectMeta.Annotations) (similar change for services)

Trigger for getting latest change: https://github.com/kubernetes/ingress-gce/pull/169, https://github.com/kubernetes/ingress-gce/pull/165.

cc @G-Harmon @csbell @madhusudancs

### CODE REVIEW

**Comment by nikhiljindal** [comment]
Verified that `go run test/e2e/e2e.go` is still green with these changes :)

@madhusudancs Will be great if you can take a look since I am not sure if @csbell and @G-Harmon will get to.

File: app/kubemci/pkg/gcp/backendservice/backendservicesyncer.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func (b *BackendServiceSyncer) EnsureBackendService(lbName string, ports []ingre
 	var err error
 	ensuredBackendServices := BackendServicesMap{}
 	for _, p := range ports {
-		be, beErr := b.ensureBackendService(lbName, p, hcMap[p.Port], npMap[p.Port], igLinks, forceUpdate)
+		be, beErr := b.ensureBackendService(lbName, p, hcMap[p.NodePort], npMap[p.NodePort], igLinks, forceUpdate)
```
Comment (madhusudancs) [inline app/kubemci/pkg/gcp/backendservice/backendservicesyncer.go]: This seems unrelated to latest dependencies. Is it intentional? The linked triggers don't clarify this either. 

Could you please explain this change in the PR description or link to the relevant trigger? Same question for all the `Port` -> `NodePort` changes below. I think I am missing context.

**Comment by nikhiljindal** [comment]
To add more context, the second commit has following changes that were required to fix the build:
* Rename ServicePort.Port to Nodeport: https://github.com/kubernetes/ingress-gce/pull/128
* Update the return argument from ListGlobalForwardingRules to be `[]*ForwardingRule` instead of `ForwardingRuleList`
* Update the import path for constants from ingress-gce like AppProtocol, ProtocolHTTP to be `pkg/annotations` instead of `utils`.
* Update the usage of annotations library from ingress-gce to use `annotations.FromIngress(ing)` instead of `annotations.IngAnnotations(ing.ObjectMeta.Annotations)` (similar change for services)

These are all due to changes to ingress-gce code that we are using.

File: app/kubemci/pkg/gcp/backendservice/backendservicesyncer.go (hunk 1 above)
Comment (nikhiljindal) [reply app/kubemci/pkg/gcp/backendservice/backendservicesyncer.go]: Added more context. This is because the struct that we are using from ingress-gce renamed Port to NodePort: kubernetes/ingress-gce#128

**Comment by madhusudancs** [comment]
Thanks for the clarification. It would be great if you can copy that comment to your original PR description and/or the second commit so others who might have to bisect the commits get better visibility.

**Comment by nikhiljindal** [comment]
Sure updated the PR description as well. Thanks!
